src/OpenLoco/Map/TileManager.cpp

```diff
@@ -321,9 +280,85 @@ namespace OpenLoco::Map::TileManager
 
         if (quad > 0)
         {
-            height.landHeight += quad / 2;
+            return quad / 2;
+        }
+        return 0;
+    }
+
+    /**
+     * Return the absolute height of an element, given its (x, y) coordinates
+     * remember to & with 0xFFFF if you don't want water affecting results
+     *
+     * @param x @<ax>
+     * @param y @<cx>
+     * @return height @<edx>
+     *
+     * 0x00467297 rct2: 0x00662783 (numbers different)
+     */
+    TileHeight getHeight(const Pos2& pos)
+    {
+        TileHeight height{ 16, 0 };
+        // Off the map
+        if ((unsigned)pos.x >= (Map::map_width - 1) || (unsigned)pos.y >= (Map::map_height - 1))
+            return height;
+
+        auto tile = TileManager::get(pos);
+        // Get the surface element for the tile
+        auto surfaceEl = tile.surface();
+
+        if (surfaceEl == nullptr)
+        {
+            return height;
         }
 
+        height.waterHeight = surfaceEl->water() * 16;
```

Comment:
what do the values 4 and 16 mean? just game constants?

Reply:
16 is the microZ to normalZ multiplier and 4 is the smallZ to normalZ multiplier. Water uses microZ, most tile element stuff is smallZ.

Reply:
They sound like important and common constants, are they already defined elsewhere already? Otherwise, would be nice to have constants for them

Reply:
No don't think they are defined anywhere atm. I'll do that in a follow on PR as there are a ton of places that should be looked at and refactored.

Reply:
Sounds good to me